core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java

```diff
@@ -49,21 +51,21 @@ public short getTypeCode() {
 
     @Override
     public byte[] encode() {
-        int bufferSize = msgs.size() * 1024;
-        ByteBuffer byteBuffer = ByteBuffer.allocate(bufferSize);
-        byteBuffer.putShort((short)msgs.size());
-        for (AbstractMessage msg : msgs) {
-            //msg.setChannelHandlerContext(ctx);
-            byte[] data = msg.encode();
-            byteBuffer.putShort(msg.getTypeCode());
-            byteBuffer.put(data);
+        final ByteBuf buffer = Unpooled.buffer();
```

Comment:
why do you use unpooled? maybe pooled is safer?

Reply:
make sense, but it's what Netty recommends to create a `ByteBuf`:

https://github.com/netty/netty/blob/128403b492d8490d9fc770b7a2dad669433b2790/buffer/src/main/java/io/netty/buffer/ByteBuf.java#L37-L41

and there is no something like `Pooled`, or should I call `new PooledByteBufAllocator(false).buffer()` directly? which is not recommended.

What do you think? @leizhiyuan